### DISCUSSION TITLE
Bluecat RESTful v2 API

### DISCUSSION THREAD

**Discussion by jimbju** [go-acme/lego#2218] [Q&A] [answered]
Hi,

Are there any plans for adding support for the BlueCat RESTful v2 API? Current Lego provider for BlueCat is utilizing the v1 API, which will eventually be replaced by v2.
Thank you! :)

**Comment by ldez** [comment]
Hi,

If we have the API documentation and a user to test it, we will use BlueCat API v2.

For now, I find a [migration guide](https://docs.bluecatnetworks.com/r/Address-Manager-RESTful-v2-API-Guide/v1-REST-API-to-RESTful-v2-API-migration-guide/9.6.0) but I don't find the detailled API documentation.

**Comment by jimbju** [comment]
Hi,

I have access to the BlueCat API v2 and could test.

**Reply by ldez** [reply]
I also need the API documentation: `http://{Address_Manager_IP}/api/openapi.json`

**Reply by jimbju** [reply]
[openapi.json](https://github.com/user-attachments/files/20079946/openapi.json)

**Reply by ldez** [reply]
Sorry for the late answer, but it was not easy because the API has 800 endpoints, there are a lot of non-explained elements, the migration guide doesn't help, and I didn't know the Bluecat concepts.
But after several unsuccessful tries since May 2025, I think I have something that should work now.

@jimbju could you try the PR #2791?

**Reply by jimbju** [reply]
Awesome, a colleague of mine will test and get back with results!

**Comment by ldez** [comment, marked as answer]
Closed by #2791
